tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/circuitbreaker/lifecycle/BaseCircuitBreakerOnMethod.java

```diff
@@ -0,0 +1,45 @@
+/*
+ *******************************************************************************
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package org.eclipse.microprofile.fault.tolerance.tck.circuitbreaker.lifecycle;
+
+import org.eclipse.microprofile.faulttolerance.CircuitBreaker;
+
+import javax.enterprise.context.Dependent;
+import java.io.IOException;
+import java.util.concurrent.atomic.AtomicInteger;
+
+@Dependent
+@CBLifecycle(CBLifecycleServiceType.BASE_M)
```

Comment:
I really don't like using a qualifier like this to distinguish between the test beans, it makes it much harder to navigate through the test code to find out which test is using which bean.

I realise that we already have some tests which do it this way. I assume this is because otherwise when one class extends another you get multiple beans which have the superclass as a bean type. If so, could we use `@Typed` to restrict the bean types instead.

I'd need to check it still works but I have a commit which replaces the qualifiers with `@Typed` instead.

Reply:
Here's the change I'd propose: https://github.com/Azquelt/microprofile-fault-tolerance/commit/828c2926be4ee51de216544ac242d1ad4f644eee

It does seem to work (in that all the tests fail in the expected way when run against open-liberty since we haven't implemented this yet)

Reply:
This is interesting, I didn't know about `@Typed`. I'll check.

Reply:
And yea, I copied what we do for the `@Retry` tests, even though I didn't like it. Your commit looks like a very nice simplification!

Reply:
Played with it a little and then just used your commit -- thanks!